experimental/auth-logic/wrappers/wrappers.go

```diff
@@ -96,3 +97,12 @@ func EmitAuthLogicToFile(authLogic Statement, filepath string) error {
 	_, writeErr := f.WriteString(authLogic.String())
 	return writeErr
 }
+
+// SanitizeName takes a value that was parsed from something external to
+// authorizaiton logic and returns a new value that can be used as part of the
+// syntax for principal names and arguments in the authorization logic syntax.
+// At present, it removes hyphens. This is exported because it may be used in
+// the main package to get principal names.
+func SanitizeName(oldName string) string {
+	return strings.ReplaceAll(oldName, "-", "")
```

Comment:
In general this may result in name-clashes. Is that something we should worry about, or are these names sufficiently localized?

Reply:
Is your point that there could be sources with both names like `FooBar` and `Foo-Bar` and this will map both to `FooBar` when they should have been different? If so, you are right, this could cause errors in general. It sounds pretty rare though. I guess a way to fix this is:

Define: s is a sanitizable name (SN) when `s != SanitizeName(s)` and call `SanitizeName(s)` its sanitization, then:
 - maintain a `map`, `sanMap` from sanitizations to the sanitizable names that generated them
 - whenever a sanitizable name `s` is encountered:
     - if `sanMap.get(SanitizeName(s))` is empty, add `sanMap[SanitizeName(s) --> s]`
     - else: check `sanMap.get(s) == s`
         - if equal: do nothing
         - if not equal: there is an erroneous name collision, throw error.
         
Does that solution make sense? Is there a simpler one?

Because this seems rare, I am inclined to open an issue for this rather than fix it now.

Reply:
Yes. That is what I mean. I cannot think of a simpler solution right now. Feel free to fix this in a separate PR or even  ignore it, if this situation is very unlikely to happen. But perhaps clarify in the doc comment.

Reply:
https://github.com/project-oak/transparent-release/issues/58